Integrations/src/main/java/io/deephaven/integrations/numpy/Java2NumpyCopy.java

```diff
@@ -230,6 +231,22 @@ public static void copySlice(final Table t, final long rowStart, final long[] da
         copySlice(t, rowStart, nRow, nCol, long.class, "(long)", (cs, k, idx) -> data[idx] = cs.getLong(k));
     }
 
+    /**
+     * Casts data to the desired type and copies a slice of rows into a flattened 2D array.
+     * This is useful for copying table data directly into numpy arrays.
+     *
+     * @param t        table to copy data from
+     * @param rowStart first row of data to copy
+     * @param data     array to copy data into
+     * @param nRow     number of rows to copy; also the number of rows in <code>data</code>.
+     * @param nCol     number of table columns; also the number of columns in <code>data</code>.
+     */
+    public static void copySlice(final Table t, final long rowStart, final boolean[] data, final int nRow, final int nCol) {
+        assertCopySliceArgs(t, rowStart, data == null ? -1 : data.length, nRow, nCol);
+        assert data != null;
+        copySlice(t, rowStart, nRow, nCol, long.class, "(long)", (cs, k, idx) -> data[idx] = cs.getBoolean(k));
```

Comment:
This looks sketchy.  Should it be something like:

```suggestion
        copySlice(t, rowStart, nRow, nCol, boolean.class, "(boolean)", (cs, k, idx) -> data[idx] = cs.getBoolean(k));
```

Reply:
Hmm... seems like the commented out block below is related to this.  If what you have is correct, you need to:
1) add a comment to explain what is going on -- this would be the textbook example for needing a comment
2) remove the commented out stuff below